src/functions.js

```diff
@@ -1841,6 +1841,39 @@ const functions = (() => {
         return undefined;
     }
 
+    /**
+     *
+     * @param {*} [value] - the input to which the type will be checked
+     * @returns {string} - the type of the input
+     */
+    function typeOf(value) {
+        if (value === undefined) {
+            return undefined;
+        }
+
+        if (value === null) {
+            return 'null';
+        }
+
+        if (isNumeric(value)) {
+            return 'number';
+        }
+
+        if(['string', 'boolean'].includes(typeof value)) {
+            return typeof value;
+        }
```

Comment:
Array.includes() is an ES6 function, which means that the transpiled jsonata-es5.js will not work in ES5 environments such as IE unless we explicitly put the polyfill into [polyfill.js](https://github.com/jsonata-js/jsonata/blob/master/polyfill.js).  
However, in this case, I think it would be easier to remove this block and replace it with two `if` statements for the string and boolean cases.